packages/react/src/components/TextArea/TextArea.js

```diff
@@ -45,11 +52,12 @@ const TextArea = React.forwardRef(function TextArea(
       }
     },
     ref,
+    ...(enableCounter && { maxLength: maxCount }),
```

Comment:
@abbeyhrt @laurenmrice  enabled counter being false but max count still enabled fix here

Reply:
Awesome, thanks for fixing that! This definitely works but conditional spreads like this can be a bit hard to debug later on, would you mind instead, somewhere after `textAreaProps` is defined doing something like: 
```js
if (enableCounter) {
  textareaProps.maxLength = maxCount;
}
```
?

Reply:
sure! and pushed! @abbeyhrt !